virtualizarr/tests/test_writers/test_kerchunk.py

```diff
@@ -31,7 +29,7 @@ def test_accessor_to_kerchunk_dict(self):
             "refs": {
                 ".zgroup": '{"zarr_format":2}',
                 ".zattrs": "{}",
-                "a/.zarray": '{"shape":[2,3],"chunks":[2,3],"dtype":"<i8","fill_value":null,"order":"C","compressor":null,"filters":null,"zarr_format":2}',
+                "a/.zarray": '{"shape":[2,3],"chunks":[2,3],"fill_value":0,"order":"C","filters":[],"dimension_separator":".","compressor":null,"attributes":{},"zarr_format":2,"dtype":"<i8"}',
```

Comment:
[If the fill_value passed to ArrayV3Metadata is `None`, ArrayV3Metadata will automatically coerce the fill_value to the default of the data type](https://github.com/zarr-developers/zarr-python/blob/v3.0.2/src/zarr/core/metadata/v3.py#L271-L272). For example, fill_value will be `0` if `None` is passed. This means that writing back to v2 metadata will generate a fill_value of the default for the data type, rather than set it as `null`, which is valid for Zarr V2. See also https://github.com/pydata/xarray/issues/5475#issuecomment-2383046633

Reply:
IIUC based on our discussion in https://github.com/zarr-developers/VirtualiZarr/issues/414 and https://github.com/pydata/xarray/issues/5475#issuecomment-2383046633.  This is the behavior we want and expect.  The v3 semantics are that `fill_value` denotes uninitialized regions of the array so it seems that returning a value with a consistent dtype would probably be what we want.